### PR TITLE
iRODS Xml-protocol compatibility

### PR DESCRIPTION
**Pull request by d-w-moore**
Allow special characters in object names, including XML-special entities like` & < > " ' `etc

### CODE REVIEW

File: irods/test/data_obj_test.py [hunk 1]
```diff
@@ -1463,6 +1465,49 @@ def test_register_with_checksum(self):
...
+        try:
+
+            #import pdb;pdb.set_trace() # dwm_
+
+            fname = re.sub( r'[/\n]', '',
+                            ''.join(map(chr,range(1,128))) )
+            prefix_ = 'object_'  + fname
+            resc_name = 'regWithNonPrintableNamesResc'
+            vault = make_ufs_resc_in_tmpdir(self.sess, resc_name, allow_local = loc_server)
+            def enter_file_into_irods ( session, filename, method, **options ):
```
Comment (korydraughn) [inline irods/test/data_obj_test.py:1485]: whitespace before opening parens here, but no whitespace in others.

is that on purpose?

File: irods/test/data_obj_test.py [hunk 2]
```diff
@@ -1463,6 +1465,49 @@ def test_register_with_checksum(self):
...
+                psr = ET( XML_Parser_Type.QUASI_XML, session.server_version)
+                bound_method = getattr(session.data_objects,method)
+                basename = os.path.basename(filename)
+                logical_path = '/{0.zone}/home/{0.username}/{basename}'.format(session,**locals())
+                bound_method( os.path.abspath(filename), logical_path , **options )
+                d = session.data_objects.get(logical_path) #import pprint;pprint.pprint(vars(d)) # dwm_
+                print ('Truth = ',d.path == logical_path)
```
Comment (korydraughn) [inline irods/test/data_obj_test.py:1492]: formatting in regards to comma-separated arguments is a little inconsistent.

for example, the commas in lines 1490 and 1492

**Review by korydraughn — APPROVED** [review]
this looks fine to me.

**Comment by d-w-moore** [comment]
Oops , push outdated your review comments @korydraughn , ... sorry. They probably still apply, I will check.

File: irods/test/data_obj_test.py (hunk 1 above)
Comment (d-w-moore) [reply irods/test/data_obj_test.py]: corrected

File: irods/test/data_obj_test.py (hunk 2 above)
Comment (d-w-moore) [reply irods/test/data_obj_test.py]: corrected

**Comment by d-w-moore** [comment]
Tested and ready for merge.

**Comment by trel** [comment]
do you want five commits?  or should i merge https://github.com/irods/python-irodsclient/pull/172 first (there's one conflict in the imports there now)?  and then your work on top of that?

**Comment by d-w-moore** [comment]
I guess it makes the most sense to merge #172 first. Then I'll rebase (& most likely squash) and #

**Comment by trel** [comment]
Okay, I'll work to get that one mergeable/merged, and then you can rebase here.

**Comment by trel** [comment]
okay, i merged #172, please rebase this cleanly, confirm it's good, and then squash to taste.    Let's wait to add # after one more review.

**Comment by d-w-moore** [comment]
This is squashed to a useful minimum. Awaiting review for validation and # ...

**Comment by trel** [comment]
if you're happy with this (passing tests), yes, please # em.

**Comment by d-w-moore** [comment]
Passing tests - ready to merge

**Pull request merged**